docs/DeveloperGuide.md

```diff
@@ -1,33 +1,373 @@
+
+![Bookmark Class Diagram](Images/BookmarkDG/BookmarkRun_Class.png)
+
+API: java.bookmarkRun
+
+The bookmark component consists of six major classes: `BookmarkRun`, `BookmarkStorage`, `BookmarkUi`, 
+`BookmarkCategory`, `BookmarkList` and `BookmarkParser`. 
+As shown in figure 1, `BookmarkRun` is 
+* The main class to be called when the bookmark mode is accessed.
+* The main class to access other bookmark classes.
+
+Given below, Figure 2, is the sequence diagram of how the classes interact with each 
+other when bookmark mode is accessed from the main function.
+
+![Bookmark Sequence Diagram](Images/BookmarkDG/sequence_bookmark.png)
+
+The bookmark component has two modes: the main bookmark mode and the category mode. 
+As shown in Figure 2, when `BookmarkRun` is called, 
+`BookmarkParser` will be called and return a `BookmarkCommand`. 
+Afterwards, `BookmarkRun` will then call `executeCommand` in `BookmarkCommand` which executes 
+the intended actions 
+Then, it will call `getCategorymode` to get the current mode the user is in. 
+
+**Bookmark Implementation**
+
+A more detailed explanation of `BookmarkCommand`, `BookmarkCategory` and `BookmarkList` will be 
+shown below. `BookmarkUi` and `BookmarkStorage` follow the same design implementations as the main 
+architecture. The below figure shows the command classes available and they are called based on the 
+`BookmarkParser` class as illustrated in Figure 2. 
+
+![BookmarkCommand ClassDiagram](Images/BookmarkDG/bookmarkCommand_Class.png)
+
+Figure 4 shows a more detailed sequence diagram of how the `BookmarkCategory`, `BookmarkCommand` 
+and `BookmarkList` interacts with each other for the scenario when the user input an `AddLinkCommand` 
+in Bookmark mode.
+
+![BookmarkCommand ClassDiagram](Images/BookmarkDG/AddCommand_sequenceDiagram.png)
+
+###Timetable Component
+
+This section will describe in detail how some features inside the timetable section have been 
+implemented.
+
+![Timetable Class Diagram](Images/TimeTable%20class%20diagram.png)
+
+**API:** <span style="color:blue">`TimeTableRun`.
+
+The timetable component consists of 7 major classes as shown. The above figure illustrates the
+ association and the multiplicity of the classes. 
+ 
+ 1. As shown in the figure, `TimetableRun` is the main class to be accessed 
+ when the timetable function is called. 
+ 1. It associates with the `TimeTableStorage` class which is used to save data into a text file 
+ 1. `DateList` class that contains a number of `EventList`. 
+ 1. `EventList` contains a number of `Event`. 
+ 1. `Event` class is the abstract parent class for `Lesson` and `Activity`. 
+ 1. `Event` class also contains a number of `Duration` and it has a dependency on the `EvenType` enum.
+
+The figure below is the sequence diagram of how the classes interact with each other when 
+the main function make the run(command) API call.
+
+![timetable_sequence diagram](Images/timetable_sequence%20diagram.png)
```

Comment:
The commandParser arrow in the sequence diagram is not pointing to either a construct or a activation bar, so its not very clear.